src/transform/src/reduction.rs

```diff
@@ -246,15 +246,16 @@ impl FoldConstants {
 
                     // We can fold all constant inputs together, but must apply the constraints to restrict them.
                     // We start with a single 0-ary row.
-                    let mut old_rows = vec![(repr::Row::pack::<_, Datum>(None), 1)];
-                    let mut row_packer = repr::RowPacker::new();
+                    let mut old_rows = vec![(Row::pack::<_, Datum>(None), 1)];
+                    let mut row_packer = Row::default();
                     for input in inputs.iter() {
                         if let MirRelationExpr::Constant { rows: Ok(rows), .. } = input {
                             let mut next_rows = Vec::new();
                             for (old_row, old_count) in old_rows {
                                 for (new_row, new_count) in rows.iter() {
+                                    row_packer.extend(old_row.iter().chain(new_row.iter()));
                                     next_rows.push((
-                                        row_packer.pack(old_row.iter().chain(new_row.iter())),
```

Comment:
I if we had a `length_in_bytes` visible (I guess the "for debugging only" `data` does this?) we could skip the packer here.